src/Aspire.Hosting.Azure/Bicep/AzureBicepPostgresResource.cs

```diff
@@ -12,19 +12,25 @@ namespace Aspire.Hosting.Azure;
 /// <param name="name">The name of the resource.</param>
 /// <param name="username">A delegate to resolve the username.</param>
 /// <param name="password">A delegate to resolve the password.</param>
-public class AzureBicepPostgresResource(string name, Func<string> username, Func<string> password) :
+public class AzureBicepPostgresResource(string name, string username, ParameterResource password) :
     AzureBicepResource(name, templateResouceName: "Aspire.Hosting.Azure.Bicep.postgres.bicep"),
     IResourceWithConnectionString
 {
     internal List<string> Databases { get; } = [];
 
+    /// <summary>
+    /// Gets the connection template for the manifest for the Azure Postgres Flexible Server.
+    /// </summary>
+    public string ConnectionStringExpression =>
+        $"Host={{{Name}.outputs.pgfqdn}};Username={username};Password={password.ValueExpression}";
```

Comment:
I'm not sure I like these expression strings being on the public API surface. Seems like an internal implementation detail.

Reply:
Why is GetConnectionString public but this isn't?